Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2_test.py

```diff
@@ -177,7 +139,197 @@ def test_build_readpdf_entry_object_empty_extract(mocker):
+    (2, 'https://test8.com/')
+])
+def test_get_urls_and_emails_from_pdf_file_with_encrypt(pdf_encoding_type, expected_output):
+    """
+    This test verifies URL and Emails extraction from an encrypted PDF file.
+
+        Given:
+        A number representing a type of a pdf encoding and an expected output - URL or Email address:
+        (the pdf is encrypted)
+
+            1. A PDF encoded with the first encoding, a text url ended with a slash /.
+
+            2. A PDF encoded with the first encoding, a text url ended without a slash /.
+
+            3. A PDF encoded with the first encoding, a text url without the http prefix.
+
+            4. A PDF encoded with the first encoding, a text email address.
+
+            5. A PDF encoded with the first encoding, a text url with an https prefix, and an email address in it.
+
+            6. A PDF encoded with the first encoding, a text hyperlink of a url.
+
+            7. A PDF encoded with the first encoding, aa embedded url (a url the is hyperlinked to an image).
+
+            8. A PDF encoded with the second encoding, a text url ended with a slash /.
+
+            9. A PDF encoded with the second encoding, a text url ended without a slash /.
+
+            10. A PDF encoded with the second encoding, a text url without the http prefix.
+
+            11. A PDF encoded with the second encoding, a text email address.
+
+            12. A PDF encoded with the second encoding, a text url with an https prefix, and an email address in it.
+
+            13. A PDF encoded with the second encoding, a text hyperlink of a url.
+
+            14. A PDF encoded with the second encoding, aa embedded url (a url the is hyperlinked to an image).
+
+        When:
+            Running 'get_urls_and_emails_from_pdf_file' function on the PDF file.
+
+        Then:
+            Verify that the expected amount of URLs and Email addresses was extracted from the PDF.
+            Verify that the specific URL or Email was extracted successfully.
+
+    """
+    from ReadPDFFileV2 import get_urls_and_emails_from_pdf_file, decrypt_pdf_file
+
+    # Select the PDF encoding:
+    if pdf_encoding_type == 1:  # first encoding type (a pdf created with google docs)
+        file_path = f'{CWD}/URLs_Extraction_Test_PDF_Encoding_Google_Docs_Renderer_protected.pdf'
+    else:  # second encoding type (a pdf created with mac os notes)
+        file_path = f'{CWD}/URLs_Extraction_Test_PDF_Encoding_Quartz_PDFContext_protected.pdf'
+
+    # Decrypt the PDF:
+    dec_file_path = f'{CWD}/decrypted.pdf'
+    decrypt_pdf_file(file_path, '123456', dec_file_path)
+
+    # Extract URLs and Emails:
+    urls = get_urls_and_emails_from_pdf_file(dec_file_path)
+
+    # Delete Decrypted file:
+    if os.path.exists(dec_file_path):
+        os.remove(dec_file_path)
+
+    assert len(urls) == 7
+    assert expected_output in urls
+
+
+@pytest.mark.parametrize('pdf_encoding_type, expected_output', [
+    (1, 'https://test1.com/'),
```

Comment:
can we switch this to file name to set of urls/emails that should be extracted?

Reply:
But then we will have only one test instead of 14.... would be more difficult to separate the specific case that isn't working.

Reply:
I dont really think it would be difficult to seperate, and I think the test would be clearer

Reply:
I still think that each case is a separate test. I did it with parametrize to prevent code duplication... let's talk about it.